src/main/java/com/epam/grid/engine/provider/host/slurm/SlurmHostProvider.java

```diff
@@ -0,0 +1,98 @@
+ *  * distributed under the License is distributed on an "AS IS" BASIS,
+ *  * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  * See the License for the specific language governing permissions and
+ *  * limitations under the License.
+ *
+ *
+ */
+
+package com.epam.grid.engine.provider.host.slurm;
+
+import com.epam.grid.engine.cmd.GridEngineCommandCompiler;
+import com.epam.grid.engine.cmd.SimpleCmdExecutor;
+import com.epam.grid.engine.entity.CommandResult;
+import com.epam.grid.engine.entity.EngineType;
+import com.epam.grid.engine.entity.HostFilter;
+import com.epam.grid.engine.entity.Listing;
+import com.epam.grid.engine.entity.host.Host;
+import com.epam.grid.engine.entity.host.slurm.SlurmHost;
+import com.epam.grid.engine.mapper.host.slurm.SlurmHostMapper;
+import com.epam.grid.engine.provider.host.HostProvider;
+import com.epam.grid.engine.provider.utils.CommandsUtils;
+import com.epam.grid.engine.provider.utils.slurm.host.ScontrolShowNodeParser;
+import lombok.RequiredArgsConstructor;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.context.Context;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * This is the implementation of the host provider for SLURM.
+ *
+ * @see com.epam.grid.engine.provider.host.HostProvider
+ */
+@Slf4j
+@Service
+@RequiredArgsConstructor
+@ConditionalOnProperty(name = "grid.engine.type", havingValue = "SLURM")
+public class SlurmHostProvider implements HostProvider {
+
+    private static final String FILTER = "filter";
+    private static final String SCONTROL_COMMAND = "scontrol";
+
+    private final SimpleCmdExecutor simpleCmdExecutor;
+    private final GridEngineCommandCompiler commandCompiler;
+    private final SlurmHostMapper slurmHostMapper;
+
+    @Override
+    public EngineType getProviderType() {
+        return EngineType.SLURM;
+    }
+
+    @Override
+    public Listing<Host> listHosts(final HostFilter hostFilter) {
+        final Context context = new Context();
+        context.setVariable(FILTER, hostFilter);
+        final String[] hostCommand = commandCompiler.compileCommand(getProviderType(), SCONTROL_COMMAND, context);
+        final CommandResult commandResult = simpleCmdExecutor.execute(hostCommand);
+        if (commandResult.getExitCode() != 0) {
+            CommandsUtils.throwExecutionDetails(commandResult);
+        } else if (!commandResult.getStdErr().isEmpty()) {
+            log.warn(commandResult.getStdErr().toString());
+        }
+
+        final List<String> stdOut = commandResult.getStdOut().stream()
+                .filter(ScontrolShowNodeParser::checkStdOutLine)
+                .collect(Collectors.toList());
+
+        if (stdOut.isEmpty()) {
+            CommandsUtils.throwExecutionDetails(commandResult);
```

Comment:
I believe this peace of code leads to such response, when there is no nodes:
```
{
  "time": "2022-04-27T09:46:23.35252",
  "httpStatus": "NOT_FOUND",
  "message": "No nodes in the system\n"
}
```

so it would be great to accurately `trim()` message of the error with `string.trim().replace("\n", "").replace("\r", "")`
Such method can be placed to StringUtils or something

Reply:
and `\n` `\r` can be placed to constant of course

Reply:
fixed